pkg/base/replicated.go

```diff
@@ -152,7 +157,149 @@ func renderReplicated(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (
 		}
 	}
 
-	return &commonBase, helmBases, nil
+	return &commonBase, helmBases, kotsKindsFileMap, nil
+}
+
+func renderKotsKinds(upstreamFiles []upstreamtypes.UpstreamFile, renderedConfig *kotsv1beta1.Config, renderOptions *RenderOptions, builder *template.Builder) (*kotsutil.KotsKinds, map[string][]byte, error) {
+	fileContentsMap := make(map[string][]byte)
+	kotsKinds := kotsutil.EmptyKotsKinds()
+
+	for _, upstreamFile := range upstreamFiles {
+		document := &Document{}
+		if err := yaml.Unmarshal(upstreamFile.Content, document); err != nil {
+			continue
+		}
+
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		decoded, gvk, err := decode(upstreamFile.Content, nil, nil)
+		if err != nil {
+			if document.APIVersion == "kots.io/v1beta1" || document.APIVersion == "troubleshoot.sh/v1beta2" || document.APIVersion == "velero.io/v1" {
+				return nil, nil, errors.Wrapf(err, "failed to decode kots yaml %s", upstreamFile.Path)
+			}
+			continue
+		}
+
+		switch gvk.String() {
```

Comment:
the config is a special case, so you do need to check that and not render it (like you currently do)

Reply:
I removed the rendering of the `kotsutil.KotsKinds` struct from this `renderKotsKinds` function since it wasn't being used. It can be generated with the `kotsutil.KotsKindsFromMap` function if needed after this function call and avoids writing and parsing extra files